src/main/java/jalasoft/com/models/CustomDate.java

```diff
@@ -0,0 +1,79 @@
+package jalasoft.com.models;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Objects;
+
+public class CustomDate {
+	private int year;
+	private int month;
+	private int date;
+
+	public CustomDate(int year, int month, int day) {
+		super();
+		this.year = year;
+		this.month = month;
+		this.date = day;
+		isValid(this);
+	}
+
+	public int getYear() {
+		return year;
+	}
+
+	public void setYear(int year) {
+		this.year = year;
+	}
+
+	public int getMonth() {
+		return month;
+	}
+
+	public void setMonth(int month) {
+		this.month = month;
+	}
+
+	public int getDay() {
+		return date;
+	}
+
+	public void setDay(int day) {
+		this.date = day;
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(date, month, year);
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		CustomDate other = (CustomDate) obj;
+		return date == other.date && month == other.month && year == other.year;
+	}
+
+	@Override
+	public String toString() {
+		return year + "-" + month + "-" + date;
+	}
+
+	public boolean isValid(CustomDate date) {

```

Comment:
Why do you think that it should be a method for CustomDate?

Reply:
I am using this method every time a new instance of this class is being created. That is why I thought it would be better to put it there, inside the CustomDate class. I call this method from the constructor. Maybe I am wrong, what is your opinion about the location of this method? :)

Reply:
if you are using it from the constructor, what is the purpose is to return a boolean value?

Reply:
You are right, it is not useful to return a boolean.  Instead I changed it to validate() and if the year, month or date does not match correctly, that error would  be handled with a try/catch.